src/node/progress_tracker.h

```diff
@@ -104,30 +114,27 @@ namespace ccf
           cert, bft_node_sig, tx_id.view, tx_id.seqno, node_id);
         cert.my_nonce = my_nonce;
         cert.have_primary_signature = true;
-        for (auto& sig : cert.sigs)
+        for (auto sig = cert.sigs.begin(); sig != cert.sigs.end();)
         {
           if (
-            !sig.second.is_primary &&
+            !sig->second.is_primary &&
             !store->verify_signature(
-              sig.second.node,
+              sig->second.node,
               cert.root,
-              sig.second.sig.size(),
-              sig.second.sig.data()))
+              sig->second.sig.size(),
+              sig->second.sig.data()))
           {
-            // NOTE: We need to handle this case but for now having this make a
-            // test fail will be very handy
-            throw ccf::ccf_logic_error(fmt::format(
-              "record_primary: Signature verification from {} FAILED, view:{}, "
-              "seqno:{}",
-              sig.first,
+            sig = cert.sigs.erase(sig);
```

Comment:
Can't see the type of `cert.sigs` here, but it looks like the iterator may be invalidate here.